### ISSUE TITLE
Errors computing with the full matrix semigroup

### ISSUE THREAD

**Issue by wilfwilson** [gap-packages/Semigroups#211]
I get the following errors in `unstable-3.0`. I assume this is unrelated to the other issue I posted today, since that did not involve an acting semigroup.
```
gap> S := FullMatrixSemigroup(3, 4);
<general linear monoid 3x3 over GF(2^2)>
gap> One(S) in S;
true
gap> H := GroupHClass(DClass(S, One(S)));
<Green's H-class: Matrix(GF(2^2), [[Z(2)^0, 0*Z(2), 0*Z(2)], [0*Z(2), Z(2)^0, 0*Z(2)],
   [0*Z(2), 0*Z(2), Z(2)^0]])>
gap> IsomorphismPermGroup(H);
Error, List Element: <list>[3] must have an assigned value in
  x := gens[genpos[1][i]] * trace( genpos[2][i] )
 ; at /Users/Wilf/GAP/pkg/semigroups/gap/main/lambda-rho.gi:123 called from
trace( i ); at /Users/Wilf/GAP/pkg/semigroups/gap/main/lambda-rho.gi:144 called from
LambdaOrbMult( o, m, i
 ) at /Users/Wilf/GAP/pkg/semigroups/gap/greens/greens-acting.gi:198 called from
SEMIGROUPS.RectifyLambda( H
 ); at /Users/Wilf/GAP/pkg/semigroups/gap/greens/greens-regular.gi:64 called from
SchutzenbergerGroup( H
 ) at /Users/Wilf/GAP/pkg/semigroups/gap/greens/greens-acting.gi:1614 called from
<function "unknown">( <arguments> )
brk> quit;
gap> Random(S);
Error, Semigroups: * (for matrices over a finite field): usage,
the degree or domain of the arguments do not match, at /Users/Wilf/GAP/pkg/semigroups/gap/elements/ffmat.gi:813 called from
LambdaOrbRep( o, m ) * Random( LambdaOrbSchutzGp( o, m ) )
  at /Users/Wilf/GAP/pkg/semigroups/gap/main/semigroups-acting.gi:454 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:32
```

**Comment by james-d-mitchell** [comment]
@markuspf any chance you could look at this Markus, this is in code you wrote for Semigroups 2.6, which I merged into unstable-3.0 earlier in the summer.

**Comment by james-d-mitchell** [comment]
No, wait, I think I fixed it
